frontend/workflows/projectCatalog/src/details/index.tsx

```diff
@@ -1,39 +1,122 @@
+import GroupIcon from "@material-ui/icons/Group";
+import { capitalize } from "lodash";
+
+import type { DetailWorkflowProps } from "..";
+
+import fetchProject from "./info/resolver";
+import type { ProjectInfo } from "./info/types";
+import type { DetailsCard } from "./card";
+import { MetaCard } from "./card";
+import { ProjectDetailsContext } from "./context";
+import ProjectHeader from "./header";
+import ProjectInfoCard from "./info";
+import QuickLinksCard from "./quick-links";
+
+type CardTyping = React.ReactNode | DetailsCard;
 
 const StyledContainer = styled(Grid)({
   padding: "20px",
 });
 
-const Details = () => (
-  <StyledContainer container direction="row" wrap="nowrap">
-    {/* Column for project details and header */}
-    <Grid container item direction="column">
-      <Grid item style={{ marginBottom: "22px" }} />
-    </Grid>
-    {/* Column for project quick links */}
-    <Grid container item direction="column" xs={1}>
-      <Card>
-        <Grid
-          container
-          item
-          direction="column"
-          alignItems="center"
-          spacing={1}
-          style={{ padding: "7px 0" }}
-        >
-          <Grid item>
-            <FontAwesomeIcon icon={faGithub} size="3x" />
+const DisabledItem = ({ name }: { name: string }) => (
+  <Grid item>
+    <Tooltip title={`${capitalize(name)} is disabled`}>
+      <FontAwesomeIcon icon={faLock} size="lg" />
+    </Tooltip>
+  </Grid>
+);
+
+const Details: React.FC<DetailWorkflowProps> = ({ children }) => {
+  const { projectId } = useParams();
+  const [projectInfo, setProjectInfo] = React.useState<ProjectInfo | null>(null);
+  const [metaCards, setMetaCards] = React.useState<CardTyping[]>([]);
+  const [dynamicCards, setDynamicCards] = React.useState<CardTyping[]>([]);
+
+  React.useEffect(() => {
+    if (children) {
+      const tempMetaCards: CardTyping[] = [];
+      const tempDynamicCards: CardTyping[] = [];
+
+      React.Children.forEach(children, child => {
+        if (React.isValidElement(child)) {
+          const { type } = child?.props;
+
+          switch (type) {
+            case "Metadata":
+              tempMetaCards.push(child);
+              break;
+            case "Dynamic":
+              tempDynamicCards.push(child);
+              break;
+            default: // Do nothing, invalid card
+          }
+        }
+      });
+
+      setMetaCards(tempMetaCards);
+      setDynamicCards(tempDynamicCards);
+    }
+  }, [children]);
+
+  return (
+    <ProjectDetailsContext.Provider value={{ projectInfo }}>
+      <StyledContainer container direction="row" wrap="nowrap">
+        {/* Column for project details and header */}
+        <Grid container item direction="column">
+          <Grid item style={{ marginBottom: "22px" }}>
+            {/* Static Header */}
+            {projectInfo && (
+              <ProjectHeader name={projectInfo.name} description={projectInfo.description} />
```

Comment:
We should always have the project name from the URL right?

Reply:
Updated to use projectId. The thought was that projectId could be anything for users, hash etc, so I originally had it extending from projectInfo.

Reply:
Yeah that makes sense. I think for now we can use the ID and change later if needed since the old behavior would not show anything if projectInfo didn't exist which is kind of weird when you navigate to a identified page